src/package/package.ts

```diff
@@ -37,135 +41,218 @@ const packagePrefixes = {
+   *
+   * @param connection - connection to the org
+   * @param project - instance of SfProject
+   * @param options - see {@link PackageVersionListOptions}
+   */
+  public static async listVersions(
+    connection: Connection,
+    project: SfProject,
+    options?: PackageVersionListOptions
+  ): Promise<PackageVersionListResult[]> {
+    // resolve/verify packages
+    const packages = options?.packages?.map((pkg) => {
+      const id = getPackageIdFromAlias(pkg, project);
+
+      // validate ID
+      if (id.startsWith('0Ho')) {
+        validateId(BY_LABEL.PACKAGE_ID, id);
+        return id;
+      } else {
+        throw messages.createError('errorInvalidPackageVersionId', [id]);
+      }
+    });
+    const opts = options || ({} as PackageVersionListOptions);
+    opts.packages = packages || [];
+
+    return (await listPackageVersions({ ...opts, ...{ connection } })).records;
   }
 
-  public async convert(
+  /**
+   * Convert a 1st generation package to a 2nd generation package.
+   * See {@link ConvertPackageOptions} for conversion options.
+   *
+   * @param pkgId the 1GP package ID (033) of the package to convert
+   * @param connection
+   * @param options {@link ConvertPackageOptions}
+   * @param project
+   */
+  public static async convert(
     pkgId: string,
+    connection: Connection,
     options: ConvertPackageOptions,
     project?: SfProject
   ): Promise<PackageVersionCreateRequestResult> {
-    return await convertPackage(pkgId, this.options.connection, options, project);
+    return convertPackage(pkgId, connection, options, project);
   }
 
-  public create(): Promise<void> {
-    return Promise.resolve(undefined);
-  }
-
-  public delete(): Promise<void> {
-    return Promise.resolve(undefined);
+  /**
+   * Returns the package ID of the package.
+   *
+   * @returns {string} package ID
+   */
+  public getId(): string {
+    return this.packageId;
   }
 
-  public async install(
-    pkgInstallCreateRequest: PackageInstallCreateRequest,
-    options?: PackageInstallOptions
-  ): Promise<PackageInstallRequest> {
-    return installPackage(this.options.connection, pkgInstallCreateRequest, options);
+  /**
+   * Returns the package type of the package.
+   *
+   * @returns {Promise<PackageType>}
+   */
+  public async getType(): Promise<PackageType> {
+    return (await this.getPackageData()).ContainerOptions;
   }
 
-  public async getInstallStatus(installRequestId: string): Promise<PackageInstallRequest> {
-    return getStatus(this.options.connection, installRequestId);
-  }
+  /**
+   * Returns the list of package versions for the package.
+   * See {@link PackageVersionListOptions} for list options
+   *
+   * @param options
+   * @returns {Promise<PackageVersionListResult[]>}
+   */
+  public async getPackageVersions(options?: PackageVersionListOptions): Promise<PackageVersionListResult[]> {
+    // This should be calling PackageVersion.list() here, but that method is not implemented yet.
```

Comment:
is this comment still true?

Reply:
Comment is no longer accurate - removed.